OpenRA.Mods.Common/Widgets/WorldInteractionControllerWidget.cs

```diff
@@ -128,13 +128,17 @@ public override bool HandleMouseInput(MouseInput mi)
 						var unit = World.ScreenMap.ActorsAtMouse(mousePos)
 							.WithHighestSelectionPriority(mousePos, mi.Modifiers);
 
-						if (unit != null && unit.Owner == (World.RenderPlayer ?? World.LocalPlayer))
+						// Players to be included in the selection (the viewer or all players in "Disable shroud" / "All players" mode)
+						var viewer = (World.LocalPlayer == null || World.LocalPlayer.Spectating) ? World.RenderPlayer : World.LocalPlayer;
+						var eligiblePlayers = viewer == null || (viewer.NonCombatant && viewer.Spectating) ? World.Players : new[] { viewer };
```

Comment:
Small nitpick: You can drop the parentheses here, both these lines.

Reply:
Fixed. But only on the first line. Removing the paranthesis from the second line gives a SA1408 error.